src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -527,6 +552,21 @@ public virtual Task WriteAsync(string value)
             tuple, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
         }
 
+        /// <summary>
+        /// Equivalent to WriteAsync(stringBuilder.ToString()) however it uses the
+        /// StringBuilder.GetChunks() method to avoid creating the intermediate string 
+        /// </summary>
+        /// <param name="value">The string (as a StringBuilder) to write to the stream</param>
+        public async virtual Task WriteAsync(StringBuilder value, CancellationToken cancellationToken = default)
+        {
+            if (value == null)
```

Comment:
I have seen as doing argument validation in the synchronous portion of the method, so that the invalid arguments are reported synchronously. E.g. here https://github.com/dotnet/coreclr/blob/8b72f3f583438df3a4c21861392258d0902c16b5/src/System.Private.CoreLib/shared/System/IO/Stream.cs#L131

Does it apply here?

Reply:
> Does it apply here?

Yup. https://github.com/dotnet/coreclr/pull/18281#discussion_r192773621

Reply:
It is not clear to me if a change is being requested.   We are validating the argument before the first await (thus the synchronous part of the method) already.   Does something have to change?

Reply:
All exceptions, regardless of where they come in the async method, are returned as part of the return Task rather than propagated synchronously, even if they're before the first await; doing otherwise would result in potential race conditions in various situations where sometimes an exception would emerge synchronously and sometimes as part of the Task.

I'd suggest something like:
```C#
public virtual Task WriteAsync(StringBuilder value, CancellationToken cancellationToken = default)
{
    return WriteAsyncCore(
        value ?? new ArgumentNullException(nameof(value)),
        cancellationToken);

    async Task WriteAsyncCore(StringBuilder sb, CancellationToken ct)
    {
        foreach (ReadOnlyMemory<char> chunk in sb.GetChunks())
            await WriteAsync(chunk, ct).ConfigureAwait(false);
    }
}
```